eng/common/scripts/Update-ChangeLog.ps1

```diff
@@ -0,0 +1,109 @@
+    if ($ReleaseDate)
+    {
+        try {
+            $ReleaseStatus = "({0})" -f ([System.DateTime]::ParseExact($ReleaseDate, $dateFormat, $provider)).ToString($dateFormat)
+        }
+        catch {
+            LogError "Invalid Release date. Please use a valid date in the format '$dateFormat'"
+        }
+    }
+    elseif ($Unreleased) {
+        $ReleaseStatus = "$UNRELEASED_TAG"
+    }
+    else {
+        $ReleaseStatus = "($(Get-Date -Format $dateFormat))"
+    }
+}
+
+Process {
+    $PkgProperties = Get-PkgProperties -PackageName $PackageName -ServiceDirectory $ServiceDirectory
+    $ChangeLogEntries = Get-ChangeLogEntries -ChangeLogLocation $PkgProperties.ChangeLogPath
+
+    if ($ChangeLogEntries.Count -gt 0)
+    {
+        if ($ChangeLogEntries.Contains($Version))
+        {
+            if ($ChangeLogEntries[$Version].ReleaseStatus -eq $ReleaseStatus)
+            {
+                LogWarning "Version is already present in change log with specificed ReleaseStatus [$ReleaseStatus]"
+                exit(0)
+            }
+
+            if ($Unreleased -and ($ChangeLogEntries[$Version].ReleaseStatus -ne $ReleaseStatus))
+            {
+                LogWarning "Version is already present in change log with a release date. Please review [$($PkgProperties.ChangeLogPath)]"
+                exit(0)
+            }
+
+            if (!$Unreleased -and ($ChangeLogEntries[$Version].ReleaseStatus -ne $UNRELEASED_TAG))
+            {
+                if ((Get-Date ($ChangeLogEntries[$Version].ReleaseStatus).Trim("()")) -gt (Get-Date $ReleaseStatus.Trim("()")))
+                {
+                    LogWarning "New ReleaseDate is older than existing release date in changelog. Please review [$($PkgProperties.ChangeLogPath)]"
+                    exit(0)
+                }
+            }
+        }
+
+        $PresentVersionsSorted = [AzureEngSemanticVersion]::SortVersionStrings($ChangeLogEntries.Keys)
+        $LatestVersion = $PresentVersionsSorted[0]
+
+        if ($ReplaceVersion) 
+        {
+            $NewChangeLogEntries = Edit-ChangeLogEntry -ChangeLogEntries $ChangeLogEntries -VersionToEdit $LatestVersion `
+            -NewEntryReleaseVersion $Version -NewEntryReleaseStatus $ReleaseStatus
+            Set-ChangeLogContent -ChangeLogLocation $PkgProperties.ChangeLogPath -ChangeLogEntries $NewChangeLogEntries
+        }
+        else 
+        {
+            $NewChangeLogEntries = Add-ChangeLogEntry -ChangeLogEntries $ChangeLogEntries -NewEntryVersion $Version `
+            -NewEntryReleaseStatus $ReleaseStatus
+            Set-ChangeLogContent -ChangeLogLocation $PkgProperties.ChangeLogPath -ChangeLogEntries $NewChangeLogEntries
+        }
+    }
+    else 
+    {
+        $NewChangeLogEntries = Add-ChangeLogEntry -NewEntryVersion $Version -NewEntryReleaseStatus $ReleaseStatus
+        Set-ChangeLogContent -ChangeLogLocation $PkgProperties.ChangeLogPath -ChangeLogEntries $NewChangeLogEntries
```

Comment:
Why not move the Set-ChangeLogContent function call to the end of this script instead of duplicating it in various places?